src/main/java/com/mathworks/ci/MatlabStepExecution.java

```diff
@@ -0,0 +1,101 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2019-2020 The MathWorks, Inc.
+ *  
+ */
+
+import java.io.IOException;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Launcher.ProcStarter;
+import hudson.model.Result;
+import hudson.model.TaskListener;
+
+public class MatlabStepExecution extends StepExecution implements MatlabBuild {
+    private static final long serialVersionUID = 1L;
+    private String command;
+    private EnvVars env;
+    private boolean copyScratchFile;
+
+
+    public MatlabStepExecution(StepContext context, String command, boolean copyScratchFile) {
+        super(context);
+        this.command = command;
+        this.copyScratchFile = copyScratchFile;
+    }
+
+    private String getCommand() {
+        return this.env == null ? getMatlabCommand() : this.env.expand(getMatlabCommand());
+    }
+
+    private String getMatlabCommand() {
+        return this.command;
+    }
+
+    private void setEnv(EnvVars env) {
+        this.env = env;
+    }
+
+    private EnvVars getEnv() {
+        return this.env;
+    }
+
+    @Override
+    public boolean start() throws Exception {
+        Launcher launcher = getContext().get(Launcher.class);
+        FilePath workspace = getContext().get(FilePath.class);
+        TaskListener listener = getContext().get(TaskListener.class);
+        EnvVars env = getContext().get(EnvVars.class);
+        setEnv(env);
+
+        int res = execMatlabCommand(workspace, launcher, listener, getEnv());
```

Comment:
Why is `env` being passed in here if it is a field of the class?

Reply:
removed class field as its not needed here kept  it local to method as consuming method is synchronized.